### PR TITLE
feat: implemented scan to receive

### PR DESCRIPTION
**Pull request by kdvuong**

### CODE REVIEW

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/borrow/accepted/AcceptedDetailViewModel.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            errorMessage.setValue(SCAN_ERROR.INVALID_ISBN);
+        }
+    }
+
+    // https://stackoverflow.com/questions/48699032/how-to-set-addsnapshotlistener-and-remove-in-populateviewholder-in-recyclerview
+    public void registerSnapshotListener() {
+        if (listenerRegistration == null) {
+            Request currentRequest = request.getValue();
+            if (currentRequest != null) {
+                requestRepository.getRefById(currentRequest.getId()).addSnapshotListener((snapshot, error) -> {
```
Comment (c25vdw) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/borrow/accepted/AcceptedDetailViewModel.java:87]: https://firebase.google.com/docs/firestore/query-data/listen#detach_a_listener
do we need to assign the `addSnapshotListener` return value to listenerRegistration?

Comment (kdvuong) [reply app/src/main/java/com/cmput301f20t21/bookfriends/ui/borrow/accepted/AcceptedDetailViewModel.java:87]: nice catch, yes we do, i missed it :)

**Review by c25vdw — APPROVED** [review]
lgtm

**Review by QuocTrungTran — APPROVED** [review]
lgtm

**Pull request merged**